crates/sui-config/src/genesis.rs

```diff
@@ -119,12 +118,30 @@ impl Genesis {
         0
     }
 
-    pub fn validator_set(&self) -> &[ValidatorInfo] {
-        &self.validator_set
-    }
+    pub fn validator_set(&self) -> Vec<ValidatorInfo> {
+        let mut infos = Vec::new();
+        for validator in &self.sui_system_object().validators.active_validators {
+            let info = ValidatorInfo {
+                name: validator.metadata.name.clone(),
+                account_key: AccountsPublicKey::Ed25519(validator.metadata.network_key()), //TODO this is wrong and we shouldn't have this here
```

Comment:
Curious, what is the proper way of doing this then, besides reading the correct account public key?